src/tangle.ts

```diff
@@ -81,7 +81,9 @@ export class Client<T> {
             filter(Boolean),
             scan((acc, curr) => {
                 if (typeof curr?.clients?.forEach === 'function') {
-                    curr.clients.forEach(((value, key) => acc.clients.set(key, value)));
+                    // deserialize json into es6 map
+                    const _curr: Context = { clients: new Map(curr.clients) };
```

Comment:
What is the type of `curr.clients` here?

Reply:
Now realizing the `json` in the comment here should be `array`.

Reply:
Use `Array.isArray()` instead?

Reply:
yes

Reply:
But what are we trying to achieve with this in the first place? Why do we have to transform an array into a map?

Reply:
> But what are we trying to achieve with this in the first place? Why do we have to transform an array into a map?

Because ES6 Map's are not JSON de-/serializable natively. Without this code { clients: Map } will be empty since it's being stripped by postMessage/onMessage. However, converting them to an array and back will do the trick.